### PR TITLE
scripts: Use pickle to cache the VkObject

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Currently it takes my machine `50.40 seconds` to generate all 53 files

This uses the python `pickle` module to cache the data so that we don't have to reparse the entire XML each time

This brought my machine down to `23.04` seconds

### CODE REVIEW

File: scripts/generate_source.py [hunk 1]
```diff
@@ -348,13 +354,27 @@ def RunGenerators(api: str, registry: str, grammar: str, directory: str, styleFi
...
-        reg.apiGen()
+        # The cached data is saved inside the BaseGenerator, so search for it and try
+        # to reuse the parsing for each generator file.
+        if not cacheVkObjectData:
+            if os.path.isfile(cachePath):
+                file = open(cachePath, 'rb')
+                cacheVkObjectData = pickle.load(file)
+                file.close()
+
+        if cacheVkObjectData and 'regenerate' not in generators[target]:
```
Comment (artem-lunarg) [inline scripts/generate_source.py:365]: python question: is it the check for existence of the key or the value for `generate` is also checked?

**Review by artem-lunarg — APPROVED** [review]
LGTM. One questions. Also caching is hard, hopefully this work flawlessly against all those talks. Otherwise we can add an option to disable this...

File: scripts/generate_source.py (hunk 1 above)
Comment (spencer-lunarg) [reply scripts/generate_source.py:365]: well the idea is the absence of it is `false`

**Comment by spencer-lunarg** [comment]
> LGTM. One questions. Also caching is hard, hopefully this works flawlessly against all those talks. Otherwise we can add an option to disable this...

Ya, maybe a "enable" / "opt-in" method to start would be smart... I didn't want to merge this until I could test it more places, but that might be a better way forward

**Pull request merged**